src/steps/resource-manager/appservice/client.ts

```diff
@@ -37,4 +42,34 @@ export class AppServiceClient extends Client {
       callback,
     });
   }
+
+  public async fetchAppConfiguration(
+    name: string | undefined,
+    resourceGroup: string | undefined,
+  ): Promise<WebAppsGetConfigurationResponse | null> {
+    const serviceClient = await this.getAuthenticatedServiceClient(
+      WebSiteManagementClient,
+    );
+
+    if (!name || !resourceGroup) {
+      return null;
+    }
+
+    return serviceClient.webApps.getConfiguration(resourceGroup, name);
```

Comment:
In this client call, it's possible that users do not have the required permissions in order to fetch these settings. We definitely don't want to cause the entire step to fail in that case. See below for an example of what you can do to avoid this.

```suggestion
    try {
      const response = await serviceClient.webApps.getConfiguration(resourceGroup, name);
      return response;
    } catch (err) {
      this.logger.warn({ err }, 'Warning: unable to fetch app configuration.');
      if (isMissingGetAppConfigPermission(err)) {
        this.logger.publishEvent({ name: 'MISSING_PERMISSION', description: 'Missing permission <Microsoft.Whatever/whatever/whatever>, which is used to fetch WebApp Configuration. Please update the `JupiterOne Reader` Role in your Azure environment in order to fetch these settings for your WebApp Configuration.' });
      }
      return;
  }
```

In addition, the code above will have you hitting this endpoint for every web app, even if it's failing with an auth error. You probably could refactor to just check once if there is a permissions error. If there is, you can skip all of these calls in `index.ts::fetchApps()`.

Reply:
Does this endpoint not require `Microsoft.Web/sites/config/list/action`? I thought it did but I could be wrong.